ui/src/utils/map/lineFromStopToInfraLink.ts

```diff
@@ -64,55 +64,65 @@ const distanceToNearestPointOnFeature = (
     );
 };
 
-export const removeLineFromStopToInfraLink = (map: MaplibreGLMap) => {
-  removeLayer(map, INFRA_CONNECTION_NAME);
+export const removeLineFromStopToInfraLink = (map: MapInstance | undefined) => {
+  if (map) {
+    removeLayer(map, INFRA_CONNECTION_NAME);
+  }
 };
 
 export const addLineFromStopToInfraLink = (
-  map: MaplibreGLMap,
+  map: MapInstance | undefined,
   geometry: Geometry,
 ) => {
-  const source = map.getSource(INFRA_CONNECTION_NAME);
-  if (!source) {
-    map.addSource(INFRA_CONNECTION_NAME, {
-      type: 'geojson',
-      data: { type: 'Feature', properties: {}, geometry },
-    });
-  } else {
-    source.setData({
-      type: 'Feature',
-      properties: {},
-      geometry,
-    });
+  if (map) {
```

Comment:
Ite tykkään aina käyttää early-return tyyliä, jos vaan mahdollista.
Nyt jäin hetkeksi ettiin tän if-lauseen else-osiota, mutta eihän sitä ollutkaan olemassa, ja kuitnekin aika pitkä funkkari, niin hetken sai silmäillä kaikkiä näitä if-tasoja.

```typescript
if (!map) {
  return;
}

...
```

Reply:
Alemmassa funkkarissa tohon tyyliin onkin jo tehty